docs/src/security.md

```diff
@@ -367,3 +394,69 @@ For further detail on how `pg_ident.conf` is managed by the operator, see the
 CloudNativePG delegates encryption at rest to the underlying storage class. For
 data protection in production environments, we highly recommend that you choose
 a storage class that supports encryption at rest.
+
+## Cluster wide permissions and operator deployment
+
+### Using the manifest
+
+In the default installation for a vanilla Kubernetes cluster, the operator is
+deployed using the manifest built using `Kustomize` on the yaml files generated
+by `controller-gen` using the markers available on [Kubebuilder](https://book.kubebuilder.io/).
+Sadly, these is a bit limited for now, and we can create a manifest
+with all permissions for RoleBinding or ClusterBinding, in our case, the permissions
+are cluster wide in the default deployment, thus, we recommend to deploy the operator
+in it's own single place when using this kind of deployment and add the proper permissions
+to block any external users access to the operator namespace including access to the
+operator `ServiceAccount`.
+
+### Using Helm chart
+
+Another way it's provided to deploy the operator is Helm Charts, but for now, it will
+produce the same effect that we already have in the previous deployment method, meaning
+that all the permissions will be cluster wide in this way to deploy, so please, follow
+the same permissions recommendation.
+
+### Using OLM
+
+The last method to discuss, from a security point of view, it's the OLM, the community
+provides support to build all the files and package required to deploy the operator
+from [OperatorHub.io](https://operatorhub.io/operator/cloudnative-pg) and thus, we have
+a little bit more help on how we manage the permissions.
+
+The first things to understand it's that OLM allows the users to enable the operator
+to "watch" one, a few or all the namespaces depending on how it is deployed, this open
+the possibility to be more granular on the permissions.
+
+When using OLM the default set of permissions is set to be the two types needed,
+RoleBinding and ClusterBinding, meaning that a set of permissions will be namespaced
+and other permissions will be cluster wide.
+
+### Why are cluster wide permissions needed?
+
+Currently the operator requires cluster wide permissions only for `namespace`
+and `nodes` objects. All the other permissions can be namespaced or cluster
+wide.
+
+The reason for these mandated permissions are:
+
+- namespace: required to handle the deletion of namespaces containing Clusters.
```

Comment:
The reason for the namespace doesn't seem justified. Could you please elaborate and provide more clarity on this topic, as it is critical from a security perspective?